### PR TITLE
Hotfix/fix panel names

### PR DESCRIPTION
**Pull request by DmytroHryshyn**

### CODE REVIEW

File: src/components/webview/WebviewResolver.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export class WebviewResolver {
 				Uri.joinPath(this.__extensionPath, 'intuita-webview/build'),
 				Uri.joinPath(this.__extensionPath, 'resources'),
 			],
+			retainContextWhenHidden: true,
```
Comment (grzpab) [inline src/components/webview/WebviewResolver.ts:15]: what does that do?

File: src/components/webview/WebviewPanel.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
+import { Disposable, Uri, ViewColumn, WebviewPanel, window } from 'vscode';
+import { WebviewResolver } from './WebviewResolver';
+import { WebviewMessage } from './webviewEvents';
+import { Message, MessageBus, MessageKind } from '../messageBus';
+
+type PanelOptions = Readonly<{
+	type: string;
+	title: string;
+	showOptions:
```
Comment (grzpab) [inline src/components/webview/WebviewPanel.ts:9]: IMHO the fewer possibilities of states, the better.

I would not have `showOptions` attribute at all, but a `viewColumn` attribute and `preserveFocus: boolean`

File: src/components/webview/WebviewPanel.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+export type Options = PanelOptions & WebviewOptions;
+
+export abstract class IntuitaWebviewPanel {
+	protected __panel: WebviewPanel | null = null;
+	protected __disposables: Disposable[] = [];
+	protected __webviewMounted = false;
+
+	constructor(
+		protected readonly options: Options,
```
Comment (grzpab) [inline src/components/webview/WebviewPanel.ts:31]: Can you make it `_options`?

File: src/components/webview/WebviewPanel.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+type WebviewOptions = Readonly<{
+	extensionUri: Uri;
+	webviewName: string;
+	initialData: unknown;
+}>;
+
+export type Options = PanelOptions & WebviewOptions;
+
+export abstract class IntuitaWebviewPanel {
+	protected __panel: WebviewPanel | null = null;
```
Comment (grzpab) [inline src/components/webview/WebviewPanel.ts:26]: if it's protected, let's perhaps make it `_panel`

`x` => public
`_x` => protected
`__x` => private

does that make sense (it's loosely following the Python conventions)?

File: src/components/webview/SourceControlWebviewPanel.ts [hunk 5]
```diff
@@ -1,95 +1,24 @@
...
-						disposable?.dispose();
-						this.__webviewMounted = true;
-						resolve(null);
-					}
-				},
-			);
-		});
-
-		return initWebviewPromise;
+	public dispose() {
```
Comment (grzpab) [inline src/components/webview/SourceControlWebviewPanel.ts:19]: can you try to add `override` (it's a new TS keyword) and see if it complains should you remove `super.dispose()`?

File: src/components/webview/WebviewResolver.ts (hunk 1 above)
Comment (DmytroHryshyn) [reply src/components/webview/WebviewResolver.ts:15]: this keeps webview active when we switch to other tabs, if set to false webview will unmount each time

File: src/components/webview/WebviewPanel.ts (hunk 4 above)
Comment (DmytroHryshyn) [reply src/components/webview/WebviewPanel.ts:26]: in js its usually `_x` for private and `x` for everything else.

Comment (DmytroHryshyn) [reply src/components/webview/WebviewPanel.ts:26]: we definitely need to add this to our code style guide doc

**Review by grzpab — APPROVED** [review]

**Pull request merged**